sphinx/writers/html.py

```diff
@@ -564,17 +564,20 @@ def depart_compact_paragraph(self, node):
 
     def visit_download_reference(self, node):
         # type: (nodes.Node) -> None
+        atts = {'class': 'reference download',
+                'download': ''}
         if self.builder.download_support and node.hasattr('filename'):
-            self.body.append(
-                '<a class="reference download internal" href="%s" download="">' %
-                posixpath.join(self.builder.dlpath, node['filename']))
-            self.context.append('</a>')
+            atts['class'] += ' internal'
+            atts['href'] = posixpath.join(self.builder.dlpath, node['filename'])
         else:
-            self.context.append('')
+            atts['class'] += ' external'
+            atts['href'] = node['reftarget']
```

Comment:
This `else` clause is not correct. This block is also used for non-download-supported builders.

Reply:
Note: I have to modify html5 writer also.